src/test/java/seedu/duke/ParserTest.java

```diff
@@ -52,162 +80,172 @@ public void parse_NotProgramCommand_errorMessage() {
     @Test
     public void parse_addCommandInvalidArgs_throwsIllegalFormatException() {
         //throws IllegalFormatException in parser package: todo decide whether to merge exceptions into one package
+
         final String[] inputs = {
-            "add ",
-            "add c/books p/$37 q/1",
-            "add n/Harry Potter 1 c/books "
+            ADD_STRING,
+            ADD_STRING + WHITESPACE + CATEGORY_EXAMPLE_1 + WHITESPACE + QUANTITY_EXAMPLE_1,
+            ADD_STRING + WHITESPACE + PURCHASE_COST_EXAMPLE_1 + WHITESPACE + CATEGORY_EXAMPLE_1,
+            ADD_STRING + WHITESPACE + QUANTITY_EXAMPLE_1 + WHITESPACE + PURCHASE_COST_EXAMPLE_1,
+            ADD_STRING + WHITESPACE + CATEGORY_EXAMPLE_1 + WHITESPACE + PURCHASE_COST_EXAMPLE_1
+                    + WHITESPACE + SELLING_PRICE_EXAMPLE_1,
+            ADD_STRING + WHITESPACE + CATEGORY_EXAMPLE_1 + WHITESPACE + PURCHASE_COST_EXAMPLE_1
+                    + WHITESPACE + QUANTITY_EXAMPLE_1
         };
 
-        final String expectedErrorOutput = String.format(
-            Parser.CORRECT_COMMAND_MESSAGE_STRING_FORMAT, Parser.ADD_ITEM_DATA_ARGS_FORMAT_STRING);
         for (String input : inputs) {
             assertThrows(IllegalFormatException.class, () -> parser.parseCommand(input, list));
-            //placeholder for now: todo equals method for comparing classes
         }
     }
 
     @Test
-    public void parse_addCommandValidArgs_parsesNormally() {
-        final String[] inputs = {
-            "add n/Harry Potter 1 c/books p/$37 s/$50 q/1",
-            "add n/Pilot P100 c/stationary p/$37 s/$50 q/1 r/Not many people bought this. Consider a 50% discount."
-        };
-        for (String input : inputs) {
-            assertTrue(true); //placeholder for now: todo equals method for comparing classes
-        }
+    public void parse_addCommandValidArgs_returnsAddCommand() throws ItemNotExistException,
+            NoPropertyFoundException, IllegalFormatException {
+
+        String input1 = ADD_STRING + " n/" + ITEM_NAME_EXAMPLE_1 + " c/" + CATEGORY_EXAMPLE_1 + " p/$"
+                + PURCHASE_COST_EXAMPLE_1 + " s/$" + SELLING_PRICE_EXAMPLE_1 + " q/" + QUANTITY_EXAMPLE_1;
+
+        AddCommand expectedCommand1 = new AddCommand(ITEM_NAME_EXAMPLE_1,
+                PURCHASE_COST_EXAMPLE_1, SELLING_PRICE_EXAMPLE_1);
+
+        assertEquals(expectedCommand1.getClass(), parser.parseCommand(input1, list).getClass());
```

Comment:
Not sure if it is enough to just check that the Command returned by the parseCommand is of the same Command type with the expected Command. I was thinking if can add an equals() method in command to compare the commands in a more accurate way. But maybe can improve on the testing in a later version.

Reply:
Noted, will discuss more in the later parts.